session/session.go

```diff
@@ -92,6 +92,7 @@ type sessionImpl struct {
 	frontendID        string                      // the id of the frontend that owns the session
 	frontendSessionID int64                       // the id of the session on the frontend server
 	Subscriptions     []*nats.Subscription        // subscription created on bind when using nats rpc server
+	requestsInFlight  map[string]string           // whether the session is waiting from a response from a remote
```

Comment:
Will this be consumed by multiple goroutines, in which case we should use sync.Map?

Reply:
Added a mutex instead, simplifies counting the values and should be slightly more performant :)